mmv1/templates/terraform/pre_create/alloydb_restore_cluster.go.erb

```diff
@@ -0,0 +1,43 @@
+// Read the restore variables from obj and remove them, since they do not map to anything in the cluster
```

Comment:
My assumption is that if we implement these fields as the same structure as API, we do not need to delete and add all fields back to `obj`. However, I could easily miss something and there's a specific reason that we need to implement it this way.

Reply:
Ahh that's a good point, let me take a look and report back! Would definitely favor the nested approach if that works.

Reply:
I remember why I did this. There's a small but consequential difference between the CREATE and RESTORE APIs.

Create: http://google3/google/cloud/alloydb/v1main/service.proto;l=108;rcl=553404819
Restore: http://google3/google/cloud/alloydb/v1main/service.proto;l=255;rcl=553404819

The create API's body gets mapped to "cluster". Whereas the restore API's body gets mapped to "*". 

- For create, these fields do not map to anything in the cluster definition. Therefore we need to remove them from obj.
- For restore, I treat obj as the cluster fields (see lines 38-41 in this file). So i need to remove them from obj since they dont map to anything in the cluster. I then manually add backupSource or continuousBackupSource.

Reply:
Oh, right, the config body got mapped to `cluster` object instead for restore. yeah, it makes sense. Thanks for the clarification.